app/src/parse_json.c

```diff
@@ -2378,52 +2378,25 @@ int st_app_parse_json(st_json_context_t* ctx, const char* filename) {
-      err("%s, failed to allocate tx_audio_sessions\n", __func__);
-      ret = -ST_JSON_NULL;
-      goto error;
-    }
-    ctx->tx_anc_sessions = (st_json_ancillary_session_t*)st_app_zmalloc(
-        ctx->tx_anc_session_cnt * sizeof(st_json_ancillary_session_t));
-    if (!ctx->tx_anc_sessions) {
-      err("%s, failed to allocate tx_anc_sessions\n", __func__);
-      ret = -ST_JSON_NULL;
-      goto error;
-    }
-    ctx->tx_fmd_sessions = (st_json_fastmetadata_session_t*)st_app_zmalloc(
-        ctx->tx_fmd_session_cnt * sizeof(st_json_fastmetadata_session_t));
-    if (!ctx->tx_fmd_sessions) {
-      err("%s, failed to allocate tx_fmd_sessions\n", __func__);
-      ret = -ST_JSON_NULL;
-      goto error;
-    }
-    ctx->tx_st22p_sessions = (st_json_st22p_session_t*)st_app_zmalloc(
-        ctx->tx_st22p_session_cnt * sizeof(st_json_st22p_session_t));
-    if (!ctx->tx_st22p_sessions) {
-      err("%s, failed to allocate tx_st22p_sessions\n", __func__);
-      ret = -ST_JSON_NULL;
-      goto error;
-    }
-    ctx->tx_st20p_sessions = (st_json_st20p_session_t*)st_app_zmalloc(
-        ctx->tx_st20p_session_cnt * sizeof(st_json_st20p_session_t));
-    if (!ctx->tx_st20p_sessions) {
-      err("%s, failed to allocate tx_st20p_sessions\n", __func__);
-      ret = -ST_JSON_NULL;
-      goto error;
-    }
-    ctx->tx_st30p_sessions = (st_json_st30p_session_t*)st_app_zmalloc(
-        ctx->tx_st30p_session_cnt * sizeof(st_json_st30p_session_t));
-    if (!ctx->tx_st30p_sessions) {
-      err("%s, failed to allocate tx_st30p_sessions\n", __func__);
+    ctx->tx_video_sessions = st_app_zmalloc((size_t)ctx->tx_video_session_cnt *
+                                            sizeof(st_json_video_session_t));
+    ctx->tx_audio_sessions = st_app_zmalloc((size_t)ctx->tx_audio_session_cnt *
+                                            sizeof(st_json_audio_session_t));
+    ctx->tx_anc_sessions = st_app_zmalloc((size_t)ctx->tx_anc_session_cnt *
+                                          sizeof(st_json_ancillary_session_t));
+    ctx->tx_fmd_sessions = st_app_zmalloc((size_t)ctx->tx_fmd_session_cnt *
+                                          sizeof(st_json_fastmetadata_session_t));
+    ctx->tx_st22p_sessions = st_app_zmalloc((size_t)ctx->tx_st22p_session_cnt *
+                                            sizeof(st_json_st22p_session_t));
+    ctx->tx_st20p_sessions = st_app_zmalloc((size_t)ctx->tx_st20p_session_cnt *
+                                            sizeof(st_json_st20p_session_t));
+    ctx->tx_st30p_sessions = st_app_zmalloc((size_t)ctx->tx_st30p_session_cnt *
+                                            sizeof(st_json_st30p_session_t));
+
+    if (!ctx->tx_video_sessions || !ctx->tx_audio_sessions || !ctx->tx_anc_sessions ||
+        !ctx->tx_fmd_sessions || !ctx->tx_st22p_sessions || !ctx->tx_st20p_sessions ||
+        !ctx->tx_st30p_sessions) {
+      err("%s, failed to allocate tx sessions\n", __func__);
```

Comment:
Is it ok, that we dont know which session allocation failed? Is it not better, to verify it after each session separately and got error immediately?

Reply:
I don't think soo honesty

This solution is better IMO 
We don't access the data, the code is much cleaner and after we go into error handling the time complexity hardly matters as it will fail anyway. 
Wdyt @PanKaker  ?